arch/xtensa/src/esp32/esp32_irq.c

```diff
@@ -595,11 +580,7 @@ void up_enable_irq(int irq)
   int cpuint = IRQ_GETCPUINT(g_irqmap[irq]);
 
   DEBUGASSERT(cpuint >= 0 && cpuint <= ESP32_CPUINT_MAX);
-#ifdef CONFIG_SMP
   DEBUGASSERT(cpu >= 0 && cpu <= CONFIG_SMP_NCPUS);
```

Comment:
```suggestion
  DEBUGASSERT(cpu >= 0 && cpu < CONFIG_SMP_NCPUS);
```
Not related to the change proposed in this PR, but since `cpu` is the CPU ID, it can't be equal to 1 in a single CPU scenario.

Reply:
Yes, you are right.  `cpu` here is used to index the CPU interrupts array.  @xiaoxiang781216 can you please apply @gustavonihei's suggestion and also fix the same assertion in `up_disable_irq` in line 532.

Reply:
Done.